docs/reference/components/sinks/gcp_cloud_storage.cue

```diff
@@ -0,0 +1,173 @@
+
+  classes: {
+    commonly_used: true
+    function: "transmit"
+    service_providers: ["GCP"]
+  }
+
+  features: {
+    batch: {
+      enabled: true
+      common: false,
+      max_bytes: 10485760,
+      max_events: null,
+      timeout_secs: 300
+    }
+    buffer: enabled: true
+    compression: {
+      enabled: true
+      default: null
+      gzip: true
+    }
+    encoding: {
+      enabled: true
+      default: null
+      json: null
+      ndjson: null
+      text: null
+    }
+    healthcheck: enabled: true
+    request: {
+      enabled: true
+      in_flight_limit: 25,
+      rate_limit_duration_secs: 1,
+      rate_limit_num: 1000,
+      retry_initial_backoff_secs: 1,
+      retry_max_duration_secs: 10,
+      timeout_secs: 60
+    }
+    tls: {
+      enabled: true
+      can_enable: false
+      can_verify_certificate: true
+      can_verify_hostname: true
+      enabled_default: false
+    }
+  }
+
+  statuses: {
+    delivery: "at_least_once"
+    development: "beta"
+  }
+
+  support: {
+    input_types: ["log"]
+
+    platforms: {
+      "aarch64-unknown-linux-gnu": true
+      "aarch64-unknown-linux-musl": true
+      "x86_64-apple-darwin": true
+      "x86_64-pc-windows-msv": true
+      "x86_64-unknown-linux-gnu": true
+      "x86_64-unknown-linux-musl": true
+    }
+
+    requirements: []
+    warnings: []
+  }
+
+  configuration: {
+    acl: {
+      common: false
+      description: "Predefined ACL to apply to the created objects. For more information, see [Predefined ACLs][urls.gcs_predefined_acl]. If this is not set, GCS will apply a default ACL when the object is created."
+      required: false
+      warnings: []
+      type: string: {
+        default: null
+        enum: {
+          authenticatedRead: "Gives the bucket or object owner OWNER permission, and gives all authenticated Google account holders READER permission."
+          bucketOwnerFullControl: "Gives the object and bucket owners OWNER permission."
+          bucketOwnerRead: "Gives the object owner OWNER permission, and gives the bucket owner READER permission."
+          private: "Gives the bucket or object owner OWNER permission for a bucket or object."
+          projectPrivate: "Gives permission to the project team based on their roles. Anyone who is part of the team has READER permission. Project owners and project editors have OWNER permission. This the default."
+          publicRead: "Gives the bucket or object owner OWNER permission, and gives all users, both authenticated and anonymous, READER permission. When you apply this to an object, anyone on the Internet can read the object without authenticating."
```

Comment:
According to the code, these are in "kebab case" meaning, for example, `authenticated-read`.